### PR TITLE
[Jira plugin] Added group members endpoint

### PR DESCRIPTION
**Pull request by rupinder-goyal-glean**
This pr adds a new endpoint to get group members. I have kept the query params in line with the api provided by jira ( [documentation link ](https://docs.atlassian.com/software/jira/docs/api/REST/9.14.0/#api/2/group-getUsersFromGroup)

### CODE REVIEW

File: jira/server/scio_search/pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.askscio.atlassian_plugins.jira</groupId>
     <artifactId>glean_search</artifactId>
-    <version>1.5.0</version>
+    <version>1.6.0</version>
```
Comment (sriram-vudayagiri-glean) [inline jira/server/scio_search/pom.xml]: i see there are other changes also being planned similar to this. should we bump after all the changes are pushed?

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Inject
+    public ScioGroupMembers(GroupManager groupManager, UserManager userManager) {
+        this.groupManager = groupManager;
+        this.userManager = userManager;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public GroupMemberResponse getGroupMembers(@QueryParam("groupname") String groupName, @QueryParam("startAt") int startAt, @QueryParam("maxResults") int maxResults) {
```
Comment (sriram-vudayagiri-glean) [inline jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: do we need to escape/unescape group names? will there be groups with spaces?

also, for consistency we can rename groupname -> groupName. just group also is fine by me

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/AtlassianUser.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+public class AtlassianUser {
+
+    public String username;
+    public String name;
+    public String key;
+    public boolean active;
+
```
Comment (sriram-vudayagiri-glean) [inline jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/AtlassianUser.java]: super nit: remove spaces

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        Utils.validateUserIsAdmin(userManager);
+        PageRequest pageRequest = PageRequests.request((long)startAt, maxResults);
+        Page<ApplicationUser> users = groupManager.getUsersInGroup(groupName, false /*includeInactiveUsers*/, pageRequest);
```
Comment (sriram-vudayagiri-glean) [inline jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: should we include inactive users as well and filter it in the client code?

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java (hunk 2 above)
Comment (sriram-vudayagiri-glean) [reply jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: should we have a cap on maxResults?

**Review by sriram-vudayagiri-glean — APPROVED** [review]
LGTM, some comments

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java (hunk 4 above)
Comment (sriram-vudayagiri-glean) [reply jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: minor: if the group doesn't exist, should we return a 404?

File: jira/server/scio_search/pom.xml (hunk 1 above)
Comment (rupinder-goyal-glean) [reply jira/server/scio_search/pom.xml]: yeah, discussed the same with Bharath.

**Review by bharath-bhushan-glean — APPROVED** [review]

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java (hunk 2 above)
Comment (siddhant-mago-glean) [reply jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: Jira's API also uses `groupname`, so might be better to leave it as it is? Good to have a cap on `maxResults` (maybe 100)

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    @Inject
+    public ScioGroupMembers(GroupManager groupManager, UserManager userManager) {
+        this.groupManager = groupManager;
+        this.userManager = userManager;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public GroupMemberResponse getGroupMembers(@QueryParam("groupname") String groupName, @QueryParam("startAt") int startAt, @QueryParam("maxResults") int maxResults) {
+        Utils.validateUserIsAdmin(userManager);
```
Comment (siddhant-mago-glean) [inline jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java:40]: Are we still doing this?

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java (hunk 4 above)
Comment (siddhant-mago-glean) [reply jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: +1

**Review by siddhant-mago-glean — APPROVED** [review]

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java (hunk 5 above)
Comment (rupinder-goyal-glean) [reply jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java:40]: have changed it in a later pr.

File: jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java (hunk 4 above)
Comment (rupinder-goyal-glean) [reply jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioGroupMembers.java]: I think there is no function to check whether the group exists or not. Also the getUsersInGroup function does not throw any exception if the group is not present.

**Pull request merged**